hack/lib/log.sh

```diff
@@ -34,6 +34,7 @@ function os::log::clean_up_logger() {
     set +o errexit
 
     if jobs -pr | grep -q "${LOGGER_PID}"; then
+        pgrep -P "${LOGGER_PID}" | xargs kill -SIGTERM &> /dev/null
```

Comment:
We should investigate this. Killing children may corrupt the data and that would make the logging useless. `sar` cleans up after itself and kills it's helper `sadc` on fedora, centos, and rhel. I'm interested to know why not on osx